pkg/proc/namedpipe_other.go

```diff
@@ -0,0 +1,59 @@
+//go:build !windows
+// +build !windows
+
+package proc
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"io"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+type openOnRead struct {
+	path string
+	rd   io.ReadCloser
+}
+
+func (oor *openOnRead) Read(p []byte) (n int, err error) {
+	if oor.rd != nil {
+		return oor.rd.Read(p)
+	}
+
+	fh, err := os.OpenFile(oor.path, os.O_RDONLY, os.ModeNamedPipe)
+	if err != nil {
+		return 0, err
+	}
+
+	oor.rd = fh
+	return oor.rd.Read(p)
+}
+
+func (oor *openOnRead) Close() error {
+	defer os.Remove(oor.path)
+
+	fh, _ := os.OpenFile(oor.path, os.O_WRONLY|syscall.O_NONBLOCK, 0)
+	if fh != nil {
+		fh.Close()
+	}
+
+	return oor.rd.Close()
+}
+
+func NamedPipe() (reader io.ReadCloser, output OutputRedirect, err error) {
```

Comment:
What's the purpose of this file and the Windows variant, why not simply use `os.Pipe` across the board?

Reply:
Also, calling this a "named pipe" seems disingenuous as the caller cannot provide a name, so it's essentially the same as calling `os.Pipe` but with wrapped return values.

Reply:
[pkg/proc/gdbserial/gdbserver.go](https://github.com/go-delve/delve/blob/a4f0491d4b2570b2e0f87b0472e86498e3c1f3ef/pkg/proc/gdbserial/gdbserver.go#L490)
The file path needs to be used in the `gdbserver.LLDBLaunch()`. I can't get the file path from `os.Pipe`. 
It is called NamedPipe because it uses the Named Pipe mode. But this seems disingenuous in the `namedpipe_other` file. Maybe we can call it `Redirector`?

Reply:
Ah, ok that makes sense. Yes, I like the name `Redirector` better, I think.

Reply:
done.

Reply:
@derekparker PTAL.